lib/src/components/joystick/joystick_action.dart

```diff
@@ -116,26 +132,25 @@ class JoystickAction {
     JoystickUtils.renderControl(c, _spriteAction, _rectAction, actionPaint);
   }
 
+  @override
   void update(double dt) {
+    super.update(dt);
     if (_rectBackgroundDirection != null && _dragging) {
-      final double _radAngle = atan2(
-        _dragPosition.dy - _rectBackgroundDirection.center.dy,
-        _dragPosition.dx - _rectBackgroundDirection.center.dx,
-      );
+      final diff = _dragPosition - _rectBackgroundDirection.center.toVector2();
```

Comment:
Not related to this PR, but couldn't it be refactored so that `_rectBackgroundDirection` be a Vector2 for what it is used for?

Reply:
I will do further improvements to the joystick classes in a followup, I think converting them all to PosComp will make it much simpler. also I dont understand why there is JoyAct and JoyDir they seem to be the same